### PR TITLE
 fix : Use function and operator with aggregate function

### PR DESCRIPTION
**Pull request by assam258-5892**
Now can use function and operator with aggregate functions.

### CODE REVIEW

File: src/backend/parser/parse_clause.c [hunk 1]
```diff
@@ -3655,19 +3655,7 @@ generateGroupClause(ParseState *pstate, List **targetlist, List *sortClause)
...
-				ereport(ERROR,
-						(errcode(ERRCODE_SYNTAX_ERROR),
-						 errmsg("aggregate must exist solely"),
-						 parser_errposition(pstate,
-											exprLocation((Node *) te->expr))));
-				return NIL;
-			}
-		}
-		else
+		if (!find_agg_walker((Node *) te->expr, &factx))
```
Comment (protodef) [inline src/backend/parser/parse_clause.c]: The following case fails.
```
CREATE GRAPH g;
CREATE (:v {c: 1, v: 1, w: 2});
CREATE (:v {c: 1, v: 1, w: 3});
CREATE (:v {c: 1, v: 1, w: 3});
CREATE (:v {c: 2, v: 1, w: 4});
CREATE (:v {c: 2, v: 1, w: 5});
CREATE (:v {c: 2, v: 1, w: 5});
MATCH (n:v) RETURN n.c, sum(n.v) + n.w;
```
The `MATCH` query results the following error.
```
ERROR:  column "_.n" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: MATCH (n:v) RETURN n.c, sum(n.v) + n.w;
                                           ^
```
You can't just remove that logic.

Comment (hyunsu-choi-bitnine) [reply src/backend/parser/parse_clause.c]: Problem is "_.n".

Every other is ok.

assam=# create table public.t1 ( i1 integer, i2 integer );
CREATE TABLE
assam=# select sum(i1) + i1 from t1 group by i1;
 ?column? 
----------
(0 rows)

assam=# select sum(i1) + i2 from t1 group by i1;
ERROR:  column "t1.i2" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: select sum(i1) + i2 from t1 group by i1;
                         ^

Comment (protodef) [reply src/backend/parser/parse_clause.c]: Yes. But the error message does not help. Users don't know what `column _.n` is and there is no `GROUP BY` clause in Cypher.

Comment (hyunsu-choi-bitnine) [reply src/backend/parser/parse_clause.c]: Fix error message problem as new feature "Support group aggregation on complex expression" in future.

Comment (protodef) [reply src/backend/parser/parse_clause.c]: OK. I think it would be better to add more commits about the fix you mentioned above to this pull request.

**Pull request closed without merging**